python/cairo-addons/src/cairo_addons/hints/hashdict.py

```diff
@@ -54,7 +54,7 @@ def hashdict_write(dict_manager: DictManager, ids: VmConsts, memory: MemoryDict)
     if isinstance(dict_tracker.data, defaultdict):
         ids.dict_ptr.prev_value = dict_tracker.data[preimage]
     else:
-        ids.dict_ptr.prev_value = 0
+        ids.dict_ptr.prev_value = dict_tracker.data.get(preimage, 0)
```

Comment:
```suggestion
    ids.dict_ptr.prev_value = dict_tracker.data[preimage]
```

this should not be opinionated about the type of dict:
- either the value is here initially
- or the dict is a default dict

Reply:
also, update the rust hint accordingly

Reply:
This was overidden by the PR https://github.com/kkrt-labs/keth/pull/666
but not raising errors https://github.com/kkrt-labs/keth/actions/runs/13166488970/job/36747788554

Waiting on https://github.com/kkrt-labs/keth/pull/670/files